zerver/management/commands/reupload_emoji.py

```diff
@@ -0,0 +1,44 @@
+import io
+from typing import Any
+from urllib.parse import urljoin
+
+from django.core.management import CommandError
+from django.core.management.commands import sendtestemail
+
+from zerver.lib.emoji import get_emoji_url
+from zerver.lib.outgoing_http import OutgoingSession
+from zerver.lib.upload import upload_emoji_image
+from zerver.models import RealmEmoji, UserProfile
+
+
+class Command(sendtestemail.Command):
+    def handle(self, *args: Any, **kwargs: str) -> None:
+        session = OutgoingSession(role="reupload_emoji", timeout=3, max_retries=3)
+
+        for realm_emoji in RealmEmoji.objects.order_by("id"):
+            print(f"Processing emoji {realm_emoji.id}")
+            emoji_filename = realm_emoji.file_name
+            emoji_url = get_emoji_url(emoji_filename, realm_emoji.realm_id)
+            if emoji_url.startswith("/user_avatars"):
+                emoji_url = urljoin(realm_emoji.realm.uri, emoji_url)
+
+            print(f"Downloading {emoji_url}")
```

Comment:
Should we be trying to download the `.original` files and working from them?  My vague recollection is that we've been saving them.

Reply:
Makes sense, changed this to use `.original` first and fallback to fetching the regular `emoji_url` if that fails for some reason